src/core/utils.js

```diff
@@ -1,3 +1,113 @@
 'use strict'
 
+const CID = require('cids')
+const multihashes = require('multihashes')
+const promisify = require('promisify-es6')
+const map = require('async/map')
+
 exports.OFFLINE_ERROR = 'This command must be run in online mode. Try running \'ipfs daemon\' first.'
+
+/**
+ * Break an ipfs-path down into it's hash hash and an array of links.
+ *
+ * examples:
+ *  b58Hash -> { hash: 'b58Hash', links: [] }
+ *  b58Hash/mercury/venus -> { hash: 'b58Hash', links: ['mercury', 'venus']}
+ *  /ipfs/b58Hash/links/by/name -> { hash: 'b58Hash', links: ['links', 'by', 'name'] }
+ *
+ * @param  {String} ipfsPath An ipfs-path
+ * @return {Object}            { hash: base58 string, links: [string], ?err: Error }
+ * @throws on an invalid @param ipfsPath
+ */
+function parseIpfsPath (ipfsPath) {
+  const invalidPathErr = new Error('invalid ipfs ref path')
+
+  const matched = ipfsPath.match(/^(?:\/ipfs\/)?([^/]+(?:\/[^/]+)*)\/?$/)
+  if (!matched) {
+    throw invalidPathErr
+  }
+
+  const [hash, ...links] = matched[1].split('/')
+
+  try {
+    if (CID.isCID(new CID(hash))) {
```

Comment:
By definition a `new CID` is a CID...

Reply:
Yeah, I agree it's super weird! What we're doing is just validating that a `CID` can be constructed from `hash` and the linter prevented some other techniques so I've reverted and told eslint to ignore the line.